program/tests/harvest_rewards.rs

```diff
@@ -422,27 +420,214 @@ async fn fail_holder_rewards_invalid_data() {
     );
 }
 
-#[allow(clippy::arithmetic_side_effects)]
-#[test_case(0, 0, 0, 0)]
-#[test_case(10, 5, 10, 0)]
-#[test_case(10, 5, 10, 2)]
-#[test_case(10, 5, 10, 5)]
-#[test_case(500_000, 1_000, 50_000_000, 0)]
-#[test_case(500_000, 1_000, 50_000_000, 50_000)]
-#[test_case(500_000, 1_000, 50_000_000, 100_000)]
-#[test_case(100_000_000, 10_000_000, 50_000_000, 0)]
-#[test_case(100_000_000, 10_000_000, 50_000_000, 2_500_000)]
-#[test_case(100_000_000, 10_000_000, 50_000_000, 5_000_000)]
-#[test_case(20_000_000_000, 50_000_000, 40_000_000_000, 0)]
-#[test_case(20_000_000_000, 50_000_000, 40_000_000_000, 50_000_000)]
-#[test_case(20_000_000_000, 50_000_000, 40_000_000_000, 100_000_000)]
-#[tokio::test]
-async fn success(
+struct System {
```

Comment:
nit: maybe I'm misunderstanding the point of this struct, but how about calling it `Pool`?

Reply:
I'm envisioning these values as being for the rewards "system" as a whole. Total mint supply isn't really a "pool" value, but the mint and the pool combined comprise the system, as well as the other pieces.

Reply:
I'd probably bundle that all as pool info, but how about `RewardsSystem` then?